.github/workflows/build-pull-request.yml

```
@@ -52,35 +53,59 @@ jobs:
     with:
       cache-read-only: true
 
+  gradle-plugins:
+    uses: ./.github/workflows/reusable-gradle-plugins.yml
+    with:
+      no-build-cache: true
+
   examples:
     uses: ./.github/workflows/reusable-examples.yml
     with:
       cache-read-only: true
 
-  # markdown-link-check is not included in the PR build because links to external urls can break at
-  # any time, which can be confusing for contributors
+  # this is not a required check to avoid blocking pull requests if external links break
+  markdown-link-check:
+    uses: ./.github/workflows/reusable-markdown-link-check.yml
+
+  # this is not a required check to avoid blocking pull requests if new misspellings are added
+  # to the misspell dictionary
+  misspell-check:
+    uses: ./.github/workflows/reusable-misspell-check.yml
 
-  markdown-misspell-check:
+  required-status-check:
+    needs:
+      - assemble
+      - test
+      - smoke-test
+      - muzzle
```

Do we want to require gradle-plugins build too?
```suggestion
      - muzzle
      - gradle-plugins
```